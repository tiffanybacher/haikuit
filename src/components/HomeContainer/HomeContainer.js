import React, { Component } from 'react';
import { connect } from 'react-redux';
import Haiku from '../Haiku/Haiku';
import HaikuForm from '../HaikuForm/HaikuForm';

export class HomeContainer extends Component {
  redirect = (path) => {
    this.props.history.push(path);
  }

  renderLatestHaiku = () => {
    const latestHaiku = this.props.haikus.pop();
    let haiku;
    
    if (latestHaiku) {
      haiku = <Haiku {...latestHaiku} />
    } else {
      haiku = <Haiku />
    }

    return haiku;
  }

  render() {
    return (
      <section className="HomeContainer">
        <div className="home-card">
          <h2>Your Lastest Haiku</h2>
          {this.renderLatestHaiku()}
        </div>
        <div className="home-card">
          <h2>Write a Haiku!</h2>
          <HaikuForm redirect={this.redirect}/>
        </div>
      </section>
    );
  }
}

export const mapStateToProps = (state) => ({
  haikus: state.haikus
});

export default connect(mapStateToProps)(HomeContainer);