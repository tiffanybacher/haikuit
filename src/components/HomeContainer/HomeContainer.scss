.home-card {
  background: white;
  padding: 20px;
}

h2 {
  margin: 0 0 10px 0;
}

@media only screen and (min-width: 768px) {
  .home-card {
    font-size: 1.1em;
    padding: 30px 50px;
  }

  h2 {
    margin: 0 0 30px 0;
  }
}

@media only screen and (min-width: 1000px) {
  .home-card {
    font-size: 1.2em;
    padding: 40px 75px;
  }
}

@media only screen and (min-width: 1200px) {
  .home-card {
    font-size: 1.3em;
    padding: 50px 250px;
  }
}