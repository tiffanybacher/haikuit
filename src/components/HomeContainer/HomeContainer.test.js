import React from 'react';
import { shallow } from 'enzyme';
import HomeContainer from './HomeContainer';

describe('HomeContainer', () => {
  let mockHistory;
  let wrapper;

  beforeEach(() => {
    mockHistory = [];
    wrapper = shallow(<HomeContainer history={mockHistory} />);
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should direct to /haiku/:id', () => {
    wrapper.instance().redirect('/haiku/3');

    expect(mockHistory).toEqual(['/haiku/3']);
  });
});