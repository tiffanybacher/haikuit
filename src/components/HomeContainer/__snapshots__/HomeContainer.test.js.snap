// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeContainer should match snapshot 1`] = `
<section
  className="HomeContainer"
>
  <div
    className="home-card"
  >
    <h2>
      Write a Haiku!
    </h2>
    <Connect(HaikuForm) />
  </div>
</section>
`;
